libsql-server/src/http/admin/mod.rs

```diff
@@ -310,51 +307,31 @@ async fn handle_create_namespace<M: MakeNamespace, C: Connector>(
         if !app_state.namespaces.exists(&ns) {
             return Err(Error::NamespaceDoesntExist(ns.to_string()));
         }
-        Some(ns)
-    } else {
-        None
-    };
+
+        config.shared_schema_name = Some(ns);
+    }
+
     let dump = match req.dump_url {
         Some(ref url) => {
             RestoreOption::Dump(dump_stream_from_url(url, app_state.connector.clone()).await?)
         }
         None => RestoreOption::Latest,
     };
 
-    let bottomless_db_id = match req.bottomless_db_id {
-        Some(db_id) => NamespaceBottomlessDbId::Namespace(db_id),
-        None => NamespaceBottomlessDbId::NotProvided,
-    };
-
-    let namespace = NamespaceName::from_string(namespace)?;
-    app_state
-        .namespaces
-        .create(namespace.clone(), dump, bottomless_db_id)
-        .await?;
-
-    let store = app_state.namespaces.config_store(namespace).await?;
-    let mut config = (*store.get()).clone();
-
+    config.bottomless_db_id = req.bottomless_db_id;
     config.is_shared_schema = req.shared_schema;
-    config.shared_schema_name = shared_schema_name.as_ref().map(|x| x.to_string());
+    config.heartbeat_url = req.heartbeat_url.as_deref().map(Url::parse).transpose()?;
+    config.txn_timeout = req.txn_timeout_s.map(Duration::from_secs);
+    config.max_row_size = req.max_row_size.unwrap_or(config.max_row_size);
```

Comment:
Are you sure that this won't reset settings by accident? Previously we only had to provide fields that were supposed to be changed. Now missing fields would reset config values.